### ISSUE TITLE
Decreased predictiveness when converting integers to floats

### ISSUE THREAD

**Issue by tsmith-perchgroup** [dmlc/xgboost#10683]
I am running some experimentation on a dataset with roughly 300 features and around 300k datapoints. There are roughly 50 integer variables, representing a range of one-hot encoded, label encoded and ordinal numerical data.

When I convert all integer columns from integer to float before fitting my model, I see a significant reduction in model predictiveness on the test set. 

Can anyone shed some light on why this might be? I can't find anything when performing a web search. I'm running XGBoost 2.0.3 using the sklearn API.

**Comment by trivialfis** [comment]
Hi, could you please share a reproducible example? It's not easy to make guess based on your description.

**Comment by trivialfis** [comment]
Closing as stalled. Feel free to reopen if there's a reproducible example.
